readthedocs/vcs_support/base.py

```diff
@@ -64,8 +64,8 @@ def __init__(
         self.verbose_name = verbose_name
         self.version_type = version_type
 
-        from readthedocs.doc_builder.environments import LocalEnvironment
-        self.environment = environment or LocalEnvironment(project)
+        from readthedocs.doc_builder.environments import LocalBuildEnvironment
+        self.environment = environment or LocalBuildEnvironment(record=False)
```

Comment:
Using `record=False` here will omit recording VCS commands in the output of the build, right?

Reply:
Also, is the change from `LocalEnvironment` to `LocalBuildEnvironment` correct? We were passing the `project` as first argument previously.

Reply:
> Using record=False here will omit recording VCS commands in the output of the build, right?

Yes

> Also, is the change from LocalEnvironment to LocalBuildEnvironment correct? We were passing the project as first argument previously.

We don't need the project, it wasn't used from inside the build environment. We were always passing a build enviroment from outside, we were using a local environment to avoid recording the commands.

Reply:
In that case, if we don't record the VCS commands, won't they be shown in the build output? Like `git clone` etc?

Reply:
When doing the checkout step, we pass a build environment with `record=True`. The only place where `environment` is `None` (defaulting to a build environment with `record=False`) is in tests.

Reply:
We shouldn't default to `record=False` just because it's useful for tests. That's a bad default in the codebase, can't we just pass in the proper environment in the tests?

Having test code in the actual codebase is a code smell, and in particular this is a bad default, so I'm not convinced it's a good idea.

Reply:
Sure, I'm +1 on changing the tests to use a specific environment. But looks a change for another PR

Reply:
Let's make sure we have a TODO in the code to change this then.